File: 129-sum-root-to-leaf-numbers/129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumNumbers(TreeNode root) {
        int[] ans = new int[1];
        helper(root, root.val, ans);
        return ans[0];
    }
    
    public static void helper(TreeNode root, int num, int[] ans){
        if(root.left==null && root.right==null){
            ans[0] +=num;
            // System.out.println(num);
        }
        if(root.left!=null) helper(root.left, num*10+root.left.val, ans);
        if(root.right!=null) helper(root.right, num*10+root.right.val, ans);
    }
}

File: 129-sum-root-to-leaf-numbers/129-sum-root-to-leaf-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumNumbers = function(root) {
    let ans = [0];
    helper(root, root.val, ans);
    return ans[0];
};


var helper = function(root,  num, ans){
        if(root.left==null && root.right==null){
            ans[0] +=num;
        }
        if(root.left!=null) helper(root.left, num*10+root.left.val, ans);
        if(root.right!=null) helper(root.right, num*10+root.right.val, ans);
    }